stubs/usersettings/usersettings.pyi

```diff
@@ -0,0 +1,14 @@
+from typing import Any, TypeVar
+
+_S = TypeVar("_S")
+
+class Settings(dict[str, object]):
```

Comment:
I recommend to use `Any` instead of object here to match the return type of `__getattr__`. Otherwise code like the following will need an explicit cast or isinstance check:

```python
settings = Settings("")
settings.add_setting("x", int)
4 + settings.x  # type checks, as x is Any
4 + setting["x"]  # fails, as "x" is object
```